tools/wpt.ts

```diff
@@ -155,18 +156,31 @@ async function run() {
   console.log(`Going to run ${tests.length} test files.`);
 
   const results = await runWithTestUtil(false, async () => {
-    const results = [];
+    const results: { test: TestToRun; result: TestResult }[] = [];
+    const cores = navigator.hardwareConcurrency;
+    const inParallel = !(cores === 1 || tests.length === 1);
+    // ideally we would parallelize all tests, but we ran into some flakiness
+    // on the CI, so here we're partitioning based on the start of the test path
+    const partitionedTests = partitionTests(tests);
+
+    const iter = pooledMap(cores, partitionedTests, async (tests) => {
+      for (const test of tests) {
+        if (!inParallel) {
+          console.log(`${blue("-".repeat(40))}\n${bold(test.path)}\n`);
+        }
+        const result = await runSingleTest(
+          test.url,
+          test.options,
+          inParallel ? () => {} : createReportTestCase(test.expectation),
+          inspectBrk,
+        );
+        results.push({ test, result });
+        reportVariation(result, test.expectation);
+      }
```

Comment:
This report variation doesn't line up with any file name now. Please update the code so the file name is printed before this instead of before the file test start in case parallel is enabled

Reply:
Done.